test/x509_time_test.c

```diff
@@ -297,6 +297,56 @@ static int test_x509_cmp_time_current(void)
     return failed == 0;
 }
 
+static int test_X509_cmp_timeframe_vpm(X509_VERIFY_PARAM *vpm,
+                                       ASN1_TIME *asn1_before,
+                                       ASN1_TIME *asn1_mid,
+                                       ASN1_TIME *asn1_after)
+{
+    int always_0 = vpm != NULL
+        && (X509_VERIFY_PARAM_get_flags(vpm) & X509_V_FLAG_USE_CHECK_TIME) == 0
+        && (X509_VERIFY_PARAM_get_flags(vpm) & X509_V_FLAG_NO_CHECK_TIME) != 0;
+
+    return asn1_before != NULL && asn1_mid != NULL && asn1_after != NULL
```

Comment:
Probably should use TEST_ptr for these

Reply:
I don't think adding `TEST_ptr` here would be appropriate because these pointers/structures are not under test. The `NULL` comparisons are needed just as guards.